### ISSUE TITLE
Modifying outgoing packets without changing node_modules?

### ISSUE THREAD

**Issue by uAliFurkanY** [PrismarineJS/node-minecraft-data#233]
I'm writing a bot and I need to modify every position packet (0x14 and 0x15 to be precise) but every way I found requires me to change code in node_modules (hell) and they all work half arsed and straight up don't with control state. Thank you in advance.

**Comment by rom1504** [comment]
Do you mean the server and client you're using do not follow the Minecraft
protocol?

You could use the custom packets feature of node-minecraft-protocol

**Comment by uAliFurkanY** [comment]
No, I just want to edit the X and Z of the position packets I'm sending.

**Comment by rom1504** [comment]
Do you want to edit the type or the value ?

For the type you can use node-minecraft-protocol custom packets
For the value you can use a node-minecraft-protocol proxy, client or server depending on your use case

**Comment by uAliFurkanY** [comment]
the value, I'm running a bot

Is the proxy only way to go in this case?

**Comment by rom1504** [comment]
No if you're running a bot you can use mineflayer directly.
bot.position = new Vec3(1,2,3)

**Comment by uAliFurkanY** [comment]
but bot.setControlState('...', true) will not get filtered by that. Also I'm interacting with both mineflayer and protocol.

**Comment by rom1504** [comment]
I think you should say more about what you want to do, it's not clear.
What is "filtered" ?
You dont need to hack things around, simply implement it directly the way you want it

**Comment by iCollin** [comment]
hook outgoing packets
```
const old_write = bot._client.write.bind(bot._client);
bot._client.write = (name, data) => {
  if (['position', 'position_look'].includes(name)) {
    data.x = /*modified value*/;
    data.z = /*modified value*/;
  }
  return old_write(name, data);
}
```
